### PR TITLE
Release next js 12, Jan 31st, 2023

### PR DESCRIPTION
**Pull request by mariahosfeld**
This release will introduce next.js 12 to this repo.

**Included PRs:**
#1596 upgrade to nextjs12
#1617 tenant config language
#1618 samplePlantLocation Bug
#1619 fix: translation issues
#1620 fix too-many-filesystem-checks issue
#1624 Replace moment with date-fns
#1630 fixes failed storybook build process/Chromatic
#1641 fix: wrongly imported file
#1643 removed next-compose-plugins, fixed next.config.js 
#1654 fix translation issue for managePayout Tab
#1656 fix/donation-link layout
#1657 Add branch name checks using husky
#1662 hotfix:overlapping buttons of sample trees form
#1665 Cleanup Release - Jan 23
#1668 fix:translation issue

### CODE REVIEW

**Comment by norbertschuler** [comment]
Sorry, I have no experience with Chromatic yet (who added it?). There seems to be a problem with it now: `ERR! Error: Cannot find module 'styled-jsx'`

**Comment by mariahosfeld** [comment]
> Sorry, I have no experience with Chromatic yet (who added it?). There seems to be a problem with it now: `ERR! Error: Cannot find module 'styled-jsx'`

Thanks for taking a look, @mohitb35 will take a look as this is related to storybook.

**Comment by mariahosfeld** [comment]
Needed Bug Fixes:

- [ ] Behavior of “Click for Details” is not correct, it just zooms out
- [ ] tree v. trees (projects & profile)
- [ ] Formatting of headlines in donation link creation

**Comment by sagararyal** [comment]
#### 📝 Changed routes:


| [**/404**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/404) | [**/\[p\]**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/[p]) |
|:-:|:-:|
|<a href="https://planet-webapp-5c9bh770a-planetapp.vercel.app/404">Screenshot of /404</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/B3gLbsU4zqS3PNTJgMbVvkHlzMuMk7Zgr96VkUK8ynnBIX5EArngaMrf736gbnRCmVWMwnp9LNTVvmdFXL2p8A.png">(view full size)</a> |<a href="https://planet-webapp-5c9bh770a-planetapp.vercel.app/[p]">Screenshot of /[p]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/xy0tSjwbPwwMV55o42xjLFyCpKC5rsfraTB3_eK8M8sVKs0-5YRKkAmyTKuYfjMPVDwI-RxnvKlQ9o9D1qEEhQ.png">(view full size)</a> |



| [**/all**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/all) | [**/claim/\[type\]/\[code\]**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/claim/[type]/[code]) |
|:-:|:-:|
|<a href="https://planet-webapp-5c9bh770a-planetapp.vercel.app/all">Screenshot of /all</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/tb251yvGSufpAWE-hlw1cc-1BMKVeD-ruuj20xY6qLFlLn_5i15qdT_YwnzMmJHGABnYkCPGJso8dBgl_CZGcw.png">(view full size)</a> |<a href="https://planet-webapp-5c9bh770a-planetapp.vercel.app/claim/[type]/[code]">Screenshot of /claim/[type]/[code]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/h5g3zv-DC1FHfyHtL0Wi_jZDaKLZKTwo-hQLv6AhDOz-gOEPQvBoS4iSP4f_VLwpf07dsRQbWEEoRPFCfJ_hdQ.png">(view full size)</a> |



| [**/complete-signup**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/complete-signup) | [**/home**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/home) |
|:-:|:-:|
|<a href="https://planet-webapp-5c9bh770a-planetapp.vercel.app/complete-signup">Screenshot of /complete-signup</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/uGgXcvbiugSZEGMstdwrVJwH7UwRg_1bu7Nrpvtjyu4unqNS8qopgn32YrLFbzhbtQA9xaSTPslO6hSUiZb-jA.png">(view full size)</a> |<a href="https://planet-webapp-5c9bh770a-planetapp.vercel.app/home">Screenshot of /home</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/SUknGx9fJ99wIzfoTRQSJbZPtJKE2ZLiWpNYUyYt7lnhYNxz_B-y5B0uIrbR2fVYVoYzUNRrufCl3AfffN7YgA.png">(view full size)</a> |



And 35 other routes:
- [**/**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/)
- [**/login**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/login)
- [**/open-app**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/open-app)
- [**/profile/api-key**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/api-key)
- [**/profile/bulk-codes/\[method\]/\[id\]**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/bulk-codes/[method]/[id])
- [**/profile/bulk-codes/\[method\]**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/bulk-codes/[method])
- [**/profile/bulk-codes**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/bulk-codes)
- [**/profile/delete-account**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/delete-account)
- [**/profile/donation-link**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/donation-link)
- [**/profile/edit**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/edit)
- [**/profile/forest**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/forest)
- [**/profile/giftfund**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/giftfund)
- [**/profile/history**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/history)
- [**/profile**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile)
- [**/profile/payouts/add-bank-details**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/payouts/add-bank-details)
- [**/profile/payouts/edit-bank-details/\[id\]**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/payouts/edit-bank-details/[id])
- [**/profile/payouts**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/payouts)
- [**/profile/payouts/schedule**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/payouts/schedule)
- [**/profile/planetcash**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/planetcash)
- [**/profile/planetcash/new**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/planetcash/new)
- [**/profile/planetcash/transactions**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/planetcash/transactions)
- [**/profile/projects/\[id\]**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/projects/[id])
- [**/profile/projects**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/projects)
- [**/profile/projects/new-project**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/projects/new-project)
- [**/profile/recurrency**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/recurrency)
- [**/profile/redeem/\[code\]**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/redeem/[code])
- [**/profile/register-trees**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/register-trees)
- [**/profile/treemapper**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/treemapper)
- [**/profile/treemapper/import**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/treemapper/import)
- [**/profile/treemapper**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/treemapper)
- [**/profile/treemapper/my-species**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/treemapper/my-species)
- [**/profile/widgets**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/profile/widgets)
- [**/s/\[id\]**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/s/[id])
- [**/t/\[id\]**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/t/[id])
- [**/verify-email**](https://planet-webapp-5c9bh770a-planetapp.vercel.app/verify-email)



Commit 88ca6a15bc349ab9d59261db47718e42f997de4d (https://planet-webapp-5c9bh770a-planetapp.vercel.app).

File: src/features/common/Layout/QueryParamsContext.tsx [hunk 1]
```diff
@@ -49,7 +53,37 @@ const QueryParamsProvider: FC = ({ children }) => {
...
+        const userPreferenceLanguage =
+          navigator.language ?? navigator.languages;
+
+        //checking is user preference language matching with the application supported language
+        const languageMatched = appSupportedLocale.filter((locale) => {
+          return (
+            locale.key[0] + locale.key[1] ===
+            userPreferenceLanguage[0] + userPreferenceLanguage[1]
+          );
+        });
```
[the remark this answers is not in the document]
Comment (norbertschuler) [reply src/features/common/Layout/QueryParamsContext.tsx:74]: Will be fixes by merging https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1617.

**Comment by mohitb35** [comment]
The failed Chromatic test will be resolved by merging in #1630

File: src/features/user/BulkCodes/FormSteps.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import React, { ReactElement, SyntheticEvent } from 'react';
```
Comment (mohitb35) [inline src/features/user/BulkCodes/FormSteps.tsx:1]: This file is not used - needs to be deleted. 

Was removed in #1585

File: src/features/user/TreeMapper/MySpecies.tsx [hunk 3]
```diff
@@ -147,3 +143,34 @@ export default function MySpecies({}: Props): ReactElement {
...
+          'registerTree',
+          'tenants',
+          'treemapper',
+        ],
+        null,
+        ['en', 'de', 'fr', 'es', 'it', 'pt-BR', 'cs']
+      )),
+    },
+  };
+}
```
Comment (mohitb35) [inline src/features/user/TreeMapper/MySpecies.tsx:176]: getStaticProps seems to have been inadvertently included in this component.

**Review by mohitb35 — APPROVED** [review]
@mariahosfeld Looks good to merge

**Pull request merged**